src/goes/goes_latlon.py

```diff
@@ -79,68 +92,126 @@ def _calc_latlon(self):
         return lat_nadir, lon_nadir
 
-    def _calc_angles(self):
+    def _calc_scan_elevation_angles(self):
         """
         Calculates the scan and elevation angles from source_file_path Dataset, reshapes the scan and elevation angle
         data arrays to a single dimension, and returns a tuple containing these data arrays.
         """
-        source_dataset = Dataset(self._source_file_path, 'r')
-        source_dataset.set_auto_scale(True)
-        x = ma.getdata(source_dataset['x'][:])
-        y = ma.getdata(source_dataset['y'][:])
-        grid_x, grid_y = np.meshgrid(x, y, indexing='xy')
+        grid_x, grid_y = np.meshgrid(self._x, self._y, indexing='xy')
         scan_angle = grid_x * 180.0 / np.pi
         elevation_angle = grid_y * 180.0 / np.pi
         scan_angle = scan_angle.reshape(len(scan_angle) * len(scan_angle))
         elevation_angle = elevation_angle.reshape(len(elevation_angle) * len(elevation_angle))
-        yaw_flip_flag = source_dataset.variables['yaw_flip_flag'][0]
-        if not yaw_flip_flag:
-            scan_angle = scan_angle[::-1]
-            elevation_angle = elevation_angle[::-1]
-        source_dataset.close()
+        scan_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(scan_angle)
+        elevation_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(elevation_angle)
+        scan_angle = self._filter_by_fill_value(scan_angle)
+        elevation_angle = self._filter_by_fill_value(elevation_angle)
         return scan_angle, elevation_angle
 
+    @staticmethod
+    def _calc_scan_position(latitude):
+        """
+        Calculates the scan position...
+        """
+        scan_position = np.full(latitude.shape, 1, dtype='int32')
+        return scan_position
+
+    def _calc_sensor_zenith_azimuth_view_angles(self, latitude, longitude):
+        """
+        Calculates the sensor zenith, azimuth, and view angles. These angles are limited to values between -80 and 80
+        degrees due to a constraint in CRTM.
+        """
+        goes_imager_projection = self._source_dataset.variables['goes_imager_projection']
+        r_eq = goes_imager_projection.getncattr('semi_major_axis')
+        h = goes_imager_projection.getncattr('perspective_point_height') + \
+            goes_imager_projection.getncattr('semi_major_axis')
+        lat_0 = goes_imager_projection.getncattr('latitude_of_projection_origin')
+        lon_0 = goes_imager_projection.getncattr('longitude_of_projection_origin')
+        h_sqr = np.power(h, 2.0)
+        r_eq_sqr = np.power(r_eq, 2.0)
+        beta = np.arccos(np.cos(latitude - lat_0) * np.cos(longitude - lon_0))
+        sqrt_comp = h_sqr + r_eq_sqr - (2.0 * h * r_eq * np.cos(beta))
+        sensor_zenith_angle = np.arcsin((h * np.sin(beta)) / np.sqrt(sqrt_comp)) * 180.0 / np.pi
+        sensor_azimuth_angle = np.arcsin(np.sin(lon_0 - longitude) / np.sin(beta)) * 180.0 / np.pi
+        sensor_view_angle = sensor_zenith_angle
```

Comment:
Unless I'm missing something, This is incorrect.  

See eq. (1) of the following reference for the conversion:
https://www.researchgate.net/publication/258662879_Comparison_of_two_transmittance_algorithms_in_the_Community_Radiative_Transfer_Model_Application_to_AVHRR

For now, we neglect the index of refraction for air, so the n/n portion is 1.   `H_i` is negligible compared to the height of the satellite, so that's zero.   We may wish to update this in the future, but for now, this will get us close enough.  

The equation would thus be (please verify):  `sensor_view_angle = np.arcsin(np.sin(sensor_zenith_angle) * Rearth/(Rearth+Hsat))`, where Rearth is the nominal radius of earth in km, and Hsat is the altitude of the satellite above the surface in km.    I believe both of these parameters are provided in the metadata for the instrument.

Reply:
I could not find the correct formulation for sensor view angle and I also saw a few test data examples where they were identical. Thank you for providing the real answer for this formula! I will check out the reference and get back to you.

Reply:
The equation you reference is for sensor zenith angle and not sensor view angle. The variable theta_sat is labelled as scan angle and not sensor zenith angle. So I am slightly confused. Here are the links I used to calculate these values. https://satepsanone.nesdis.noaa.gov/pub/GASP/documentations/GOESR/GOES-R_SemiStatic_Data_README2.docx and https://www.ncdc.noaa.gov/gridsat/docs/Angle_Calculations.pdf.

Reply:
Maybe I'm confused too.   Alright, let's table this for now.   I recommend creating an issue for this specific topic, because we'll need to ensure that this part is done correctly.

Reply:
I created this issue to verify these formulations. 

https://github.com/JCSDA-internal/ioda-converters/issues/625